pkg/azdiskschedulerextender/azdiskschedulerextender.go

```diff
@@ -441,17 +438,77 @@ func setNodeScoresToZero(nodes []v1.Node) (priorityList schedulerapi.HostPriorit
 	return
 }
 
+func getDiskRequestedByPod(ns string, requestedVolumes []v1.Volume, filterOutAccessModes ...v1.PersistentVolumeAccessMode) (diskRequestedByPod map[string]struct{}, requestedDiskCount int) {
+	// create a set of all volumes needed
+	diskRequestedByPod = map[string]struct{}{}
+	for _, volume := range requestedVolumes {
+		// not the same as len(pvNamesForRequestedVolumes). PV-s for a new pod might not be cached yet.
+		if volume.PersistentVolumeClaim != nil {
+			requestedDiskCount++
+			if v, ok := pvcToPvMap.Load(getQualifiedName(ns, volume.PersistentVolumeClaim.ClaimName)); ok {
+				entry := v.(*pvcToPVEntry)
+				shouldCheckEntry := true
+				for _, filterOutAccessMode := range filterOutAccessModes {
+					if strings.EqualFold(entry.AccessMode, string(filterOutAccessMode)) {
+						shouldCheckEntry = false
+					}
+				}
+				if shouldCheckEntry {
+					if v, ok := pvToDiskNameMap.Load(entry.VolumeName); ok {
+						diskName := v.(string)
+						diskRequestedByPod[diskName] = struct{}{}
+					}
+				}
+			}
+		}
+	}
+	return
+}
+
 func onPvcUpdate(obj interface{}, newObj interface{}) {
 	// Cast the obj to pvc
 	pvc := newObj.(*v1.PersistentVolumeClaim)
-	klog.V(4).Infof("updating pvcToPv mapping cache. PVC: %s PV: %s, AccessMode: %v", pvc.Name, pvc.Spec.VolumeName, pvc.Spec.AccessModes)
-	cacheValue := cacheEntry{VolumeName: pvc.Spec.VolumeName, AccessMode: fmt.Sprintf("%v", pvc.Spec.AccessModes[0])}
-	pvcToPvMapCache.AddOrUpdate(pvc.Name, &cacheValue)
+	addPvcEntry(pvc)
+}
+
+func addPvcEntry(pvc *v1.PersistentVolumeClaim) {
+	qualifiedName := getQualifiedName(pvc.Namespace, pvc.Name)
+	klog.Infof("Updating pvcToPv map cache.PVC: %s PV: %s, AccessMode: %v", qualifiedName, pvc.Spec.VolumeName, pvc.Spec.AccessModes)
```

Comment:
The logs generated by the event callbacks should be verbose logs (i.e. `klog.V(4)` as before). This is useful for our debugging, but would be far too chatty in a customer cluster.